zerver/actions/invites.py

```diff
@@ -386,8 +388,11 @@ def do_revoke_multi_use_invite(multiuse_invite: MultiuseInvite) -> None:
     realm = multiuse_invite.referred_by.realm
 
     content_type = ContentType.objects.get_for_model(MultiuseInvite)
-    Confirmation.objects.filter(content_type=content_type, object_id=multiuse_invite.id).delete()
-    multiuse_invite.delete()
+    with transaction.atomic():
+        Confirmation.objects.filter(
+            content_type=content_type, object_id=multiuse_invite.id
+        ).delete()
+        multiuse_invite.delete()
```

Comment:
This is changed to using `MultiUseInvite.status` instead in #22920, but this here is a quick fix until we can merge that.